openvdb/openvdb/unittest/TestValueAccessor.cc

```diff
@@ -378,49 +379,42 @@ TEST_F(TestValueAccessor, testMultithreadedAccessor)
     //using AccessorT = openvdb::tree::ValueAccessor<Tree4Type>;
 
     // Task to perform multiple reads through a shared accessor
-    struct ReadTask: public tbb::task {
+    struct ReadTask {
         AccessorT& acc;
         ReadTask(AccessorT& c): acc(c) {}
-        tbb::task* execute()
+        void execute()
         {
             for (int i = -MAX_COORD; i < MAX_COORD; ++i) {
                 ASSERT_DOUBLES_EXACTLY_EQUAL(double(i), acc.getValue(openvdb::Coord(i)));
             }
-            return nullptr;
         }
     };
     // Task to perform multiple writes through a shared accessor
-    struct WriteTask: public tbb::task {
+    struct WriteTask {
         AccessorT& acc;
         WriteTask(AccessorT& c): acc(c) {}
-        tbb::task* execute()
+        void execute()
         {
             for (int i = -MAX_COORD; i < MAX_COORD; ++i) {
                 float f = acc.getValue(openvdb::Coord(i));
                 ASSERT_DOUBLES_EXACTLY_EQUAL(float(i), f);
                 acc.setValue(openvdb::Coord(i), float(i));
                 ASSERT_DOUBLES_EXACTLY_EQUAL(float(i), acc.getValue(openvdb::Coord(i)));
             }
-            return nullptr;
         }
     };
     // Parent task to spawn multiple parallel read and write tasks
-    struct RootTask: public tbb::task {
+    struct RootTask {
         AccessorT& acc;
         RootTask(AccessorT& c): acc(c) {}
-        tbb::task* execute()
+        void execute()
         {
-            ReadTask* r[3]; WriteTask* w[3];
+            tbb::task_group tasks;
             for (int i = 0; i < 3; ++i) {
-                r[i] = new(allocate_child()) ReadTask(acc);
-                w[i] = new(allocate_child()) WriteTask(acc);
+                tasks.run([&] { ReadTask r(acc); r.execute(); });
+                tasks.run([&] { WriteTask w(acc); w.execute(); });
```

Comment:
**Significant change**

This test tests multiple read/write accesses running concurrently. I believe I've ported it to represent that, even though the behaviour is not exactly identical.